generator/config/schema.json

```diff
@@ -214,8 +253,123 @@
                     "items": {
                         "type": "object"
                     }
+                },
+                "schema": {
+                    "type": [
+                        "string",
+                        "object"
+                    ],
+                    "additionalProperties": {
+                        "type": "object",
+                        "additionalProperties": {
+                            "$ref": "#/definitions/SimplifiedSchemaField"
+                        }
+                    }
                 }
             }
+        },
+        "SimplifiedSchemaType": {
+            "anyOf": [
+                {
+                    "$ref": "#/definitions/SimplifiedSchemaBaseType"
+                },
+                {
+                    "$ref": "#/definitions/SimplifiedSchemaArrayType"
+                },
+                {
+                    "$ref": "#/definitions/SimplifiedSchemaDocumentType"
+                }
+            ]
+        },
+        "SimplifiedSchemaField": {
+            "additionalProperties": false,
+            "properties": {
+                "types": {
+                    "items": {
+                        "$ref": "#/definitions/SimplifiedSchemaType"
+                    },
+                    "type": "array"
+                }
+            },
+            "required": [
+                "types"
+            ],
+            "type": "object"
+        },
+        "SimplifiedSchemaArrayType": {
+            "additionalProperties": false,
+            "properties": {
+                "bsonType": {
+                    "const": "Array",
+                    "type": "string"
+                },
+                "types": {
+                    "items": {
+                        "type": "object"
+                    },
+                    "type": "array"
+                }
+            },
+            "required": [
+                "bsonType",
+                "types"
+            ],
+            "type": "object"
+        },
+        "SimplifiedSchemaBaseType": {
+            "additionalProperties": false,
+            "properties": {
+                "bsonType": {
+                    "enum": [
+                        "Array",
+                        "Binary",
+                        "Boolean",
+                        "Code",
+                        "CodeWScope",
```

Comment:
Is the `CodeWScope` type specific to mongosh?

Reply:
This is the deprecated code_w_s bson type. Realistically, I don't expect any of the more advanced types to be found in the docs, but I generated the schema from the type definitions and opted not to manually edit it to make it easier to maintain if it ever changes. If we wanted to clean it up, we can definitely remove the deprecated types.

Reply:
@GromNaN: For additional context, PHPC [conditionally encodes Code or CodeWScope](https://github.com/mongodb/mongo-php-driver/blob/1.21.0/src/phongo_bson_encode.c#L248-L252) based on whether the MongoDB\BSON\Javascript object was constructed with a non-null `$scope` parameter (see: [`php_phongo_javascript_init`](https://github.com/mongodb/mongo-php-driver/blob/1.21.0/src/BSON/Javascript.c#L48-L52)).